core/src/main/java/gyro/core/backend/StateBackendDirectiveProcessor.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.core.backend;
+
+import java.beans.PropertyDescriptor;
+import java.lang.reflect.Method;
+
+import com.google.common.base.CaseFormat;
+import gyro.core.FileBackend;
+import gyro.core.GyroException;
+import gyro.core.Reflections;
+import gyro.core.Type;
+import gyro.core.directive.DirectiveProcessor;
+import gyro.core.scope.RootScope;
+import gyro.core.scope.Scope;
+import gyro.lang.ast.block.DirectiveNode;
+
+@Type("state-backend")
+public class StateBackendDirectiveProcessor extends DirectiveProcessor<RootScope> {
+
+    @Override
+    public void process(RootScope scope, DirectiveNode node) throws Exception {
+        validateArguments(node, 0, 2);
+        String type = getArgument(scope, node, String.class, 0);
+        String name = getArgument(scope, node, String.class, 1);
+
+        StateBackendSettings stateBackendSettings = scope.getSettings(StateBackendSettings.class);
+
+        if (stateBackendSettings.getStateBackend() != null
+            && !name.equals(stateBackendSettings.getStateBackend().getName())) {
+            throw new GyroException(node, "Only one 'state-backend' is allowed!");
+        }
```

Comment:
Rather than only allow one state backend it might be nice to allow multiple named backends and then the copy command can take a name in the `--to` and `--from` options.

If a name isn't provided, use "default". The default should be the backend used for `gyro up`.